cmd/docker-nv2/certs.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/notaryproject/nv2/cmd/docker-nv2/config"
+	"github.com/urfave/cli/v2"
+)
+
+var certsCommand = &cli.Command{
+	Name:  "certificates",
+	Usage: "Manage certificates used for signing and verification",
+	Subcommands: []*cli.Command{
+		certsAddCommand,
+		certsListCommand,
+		certsRemoveCommand,
+	},
+}
+
+var certsAddCommand = &cli.Command{
+	Name:      "add",
+	Usage:     "Add certificate to verification list",
+	ArgsUsage: "[cert]",
+
+	Action: addCert,
+}
+
+var certsListCommand = &cli.Command{
+	Name:    "list",
+	Usage:   "List certificates used for verification",
+	Aliases: []string{"ls"},
+	Action:  listCerts,
+}
+
+var certsRemoveCommand = &cli.Command{
+	Name:      "remove",
+	Usage:     "Remove certificate from verification list",
+	Aliases:   []string{"rm"},
+	ArgsUsage: "[cert]",
+	Action:    removeCerts,
+}
+
+func uniqueAppend(entries []string, e string) []string {
+	entries = append(entries, e)
+	keys := make(map[string]bool)
+	list := []string{}
+	for _, item := range entries {
+		if _, value := keys[item]; !value {
+			keys[item] = true
+			list = append(list, item)
+		}
+	}
+	return list
+}
```

Comment:
Since the input to function is slice, this method created a map from slice with slice values as keys thus removing duplicate but won't it be more efficient(both memory and compute) to just iterate over slice i.e., `entries` and check for duplicate and add `e` if none is found?

Also, by not creating intermediate map we will be preserving the order of certificates thus every invocation of `certificates ls` will provide predictable output to customer.

```
Before:
certificates add path1
certificates add path2
certificates add path3
certificates ls
> [path2, path1, path3]
certificates add path4
certificates ls
> [path2, path4, path1, path3]

After
certificates add path1
certificates add path2
certificates add path3
certificates ls
> [path1, path2, path3]
certificates add path4
certificates ls
> [path1, path2, path3, path4]
```